topics/proteomics/tutorials/neoantigen-2-non-normal-database-generation/tutorial.md

```diff
@@ -0,0 +1,666 @@
+>
+> > <solution-title></solution-title>
+> >
+> > 1. Regular expressions allow for flexible, precise pattern matching, enabling the user to find and replace complex text patterns within a column. This is useful for standardizing data formats, extracting relevant portions, or appending additional information to entries.
+> > 2. The generic|INDEL_\1 replacement pattern prepends generic|INDEL_ to each entry in the column, standardizing the label format and clarifying that each entry is related to an insertion or deletion (INDEL) variant. This makes it clear in downstream analysis that the sequence is variant-specific.
+> >
+> {: .solution}
+>
+{: .question}
+
+## Converting the manipulated tabular files to FASTA with Tabular-to-FASTA
+Tabular-to-FASTA is a tool that converts tabular data back into FASTA format, where each entry has a title and a sequence. In this step, the tool uses the processed output from the Column Regex Find And Replace step to create a FASTA file. Column c1 is set as the title (identifier) for each sequence, and column c2 contains the sequence data. This conversion is useful when the data needs to be returned to a standard sequence format for further bioinformatics analyses.
+
+In this workflow, Tabular-to-FASTA converts the formatted tabular data back into a FASTA file, making it compatible with tools that require FASTA input for further analysis. This step enables the standardized, cleaned sequences from previous steps to be utilized in additional bioinformatics workflows or databases, maintaining the variant-specific information in a commonly used format. We do this for all the tabular files (SNV, INDEL, and RPKM).
+
+> <hands-on-title> INDEL-Tabular-to-FASTA </hands-on-title>
+>
+> 1. {% tool [Tabular-to-FASTA](toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1) %} with the following parameters:
+>    - {% icon param-file %} *"Tab-delimited file"*: `out_file1` (output of **Column Regex Find And Replace** {% icon tool %})
+>    - *"Title column(s)"*: `c['1']`
+>    - *"Sequence column"*: `c2`
+>
+> 2. Rename as
+{: .hands_on}
+
+> <hands-on-title> SNV-Tabular-to-FASTA </hands-on-title>
+>
+> 1. {% tool [Tabular-to-FASTA](toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1) %} with the following parameters:
+>    - {% icon param-file %} *"Tab-delimited file"*: `out_file1` (output of **Column Regex Find And Replace** {% icon tool %})
+>    - *"Title column(s)"*: `c['1']`
+>    - *"Sequence column"*: `c2`
+>
+>
+{: .hands_on}
+
+> <hands-on-title> RPKM-Tabular-to-FASTA </hands-on-title>
+>
+> 1. {% tool [Tabular-to-FASTA](toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1) %} with the following parameters:
+>    - {% icon param-file %} *"Tab-delimited file"*: `out_file1` (output of **Column Regex Find And Replace** {% icon tool %})
+>    - *"Title column(s)"*: `c['1']`
+>    - *"Sequence column"*: `c2`
+>
+>
+{: .hands_on}
```

Comment:
since these are 3 tiny hands-on boxes without any text in between, you could consider merging them into 1 big box. Or just have the description once, and then steps 2 and 3 can just be "repeat this process for the RPKM table" etc